src/navigationGuards.ts

```diff
@@ -316,8 +316,11 @@ export function extractComponentsGuards(
               : resolved
             // replace the function with the resolved component
             record.components[name] = resolvedComponent
+            // __vccOpts is added by vue-class-component and contain the regular options
+            let options: ComponentOptions =
+              (resolvedComponent as any).__vccOpts || resolvedComponent
             // @ts-ignore: the options types are not propagated to Component
```

Comment:
this should be removed now

Reply:
I have removed the @ts-ignore , is that right?